File: Python Assignments/WeekendExercise/Task.py

def reverse_string(user_input):
    reversed_list = []

    for index in range(len(list_of_letters)):
        reversed_list.append(list_of_letters[- index + len(list_of_letters) - 1]) 
    
    for characters in reversed_list:
        print(characters, end="")
    print()

def reverse_integer(user_input):
    list_of_letters = list(user_input)
    reversed_list = []
    
    for index in range(len(list_of_letters)):
        reversed_list.append(list_of_letters[- index + len(list_of_letters) - 1]) 
    
    for numbers in reversed_list:
        print(int(numbers), end="")
    print()

def uppercase_count (list_of_letters):
    uppercase_letters = 0
    for letters in list_of_letters:
        if letters.isupper():
            uppercase_letters += 1

    print("Number of uppercase letters:", uppercase_letters)

def lowercase_count(list_of_letters):
    lowercase_letters = 0
    for letters in list_of_letters:
        if letters.islower():
            lowercase_letters += 1

    print("Number of lowercase letters:", lowercase_letters)

def first_vowel_position(list_of_letters):
    vowels = "aeiou"
    for index in range(len(list_of_letters)):
        if list_of_letters[index].lower() in vowels:
            print("The position of the first vowel in the string is", index + 1)
            break

def character_ascii (list_of_letters):
    for characters in list_of_letters:
        print(ord(characters), end=", ")
    print()

def avg_one_to_hundred():
    total = 0
    sum_total = 0
    for numbers in range(1,101):
        total += 1
        sum_total += numbers
    print("Average of numbers from 1 to 100 =",sum_total/total)

def divisors(user_input):
    print("The divisors for", user_input, "are:", end =" ")
    number_of_divisors = 0
    for index in range(1, user_input + 1):
        if user_input % index == 0:
            number_of_divisors += 1
            print(index, end=", ")
    print("\nNumber of diviorsors =",number_of_divisors)

def power(number, exp):
    result = 1
    for _ in range(exp):
        result *= number
    print(result)

def palindrome_number(user_input):
    list_of_digits = list(user_input)
    reversed_list = []

    for index in range(len(list_of_digits)):
        reversed_list.append(list_of_digits[- index + len(list_of_digits) - 1])

    if list_of_digits == reversed_list:
        print("The number is a palindrome")
    else:
        print("The number is NOT a palindrome")

def palindrome_string(user_input):
    list_of_letters = list(user_input)
    reversed_list = []

    for index in range(len(list_of_letters)):
        reversed_list.append(list_of_letters[- index + len(list_of_letters) - 1])

    if list_of_letters == reversed_list:
        print("The string is a palindrome")
    else:
        print("The string is NOT a palindrome")

def number_of_words(list_of_letters):
    number_of_words = 1
    for characters in list_of_letters:
        if characters == " ":
            number_of_words += 1
    print(number_of_words)

def sum_of_even_digits(user_input):
    sum_of_even = 0
    for numbers in user_input:
        if int(numbers) % 2 == 0:
            sum_of_even += int(numbers)
        
    print("Sum of all even digits is =", sum_of_even)

def sum_of_odd_digits(user_input):
    sum_of_odd = 0
    for numbers in user_input:
        if int(numbers) % 2 != 0:
            sum_of_odd += int(numbers)
    print("Sum of all even digits is =", sum_of_odd)

def prime_numbers():
    number_of_primes = 0
    for numbers in range(2, 101):
        is_prime = True
        for i in range (2, numbers):
            if numbers % i == 0:
                is_prime = False
                break
        if is_prime:
            number_of_primes += 1
            print(numbers, end =" ")    
            
    print("\nTotal primes between 1-100 is: ",number_of_primes)

def fibonacci_sequence():
    num_1 = 0
    num_2 =1
    for _ in range(20):
        print(num_1, end=" ")
        temp = num_2
        num_2 = num_1 + num_2
        num_1 = temp
    print()

def binary_to_decimal(user_input):
    decimal_value = 0
    for index in range(len(user_input)):
        decimal_value += int(user_input[- index - 1]) * (2 ** index)
    print("Decimal value is =", decimal_value)

def decimal_to_binary(user_input):
    binary_value = ""
    while user_input > 0:
        binary_value = str(user_input % 2) + binary_value
        user_input //= 2
    print("Binary value is =", binary_value)    

user_input = input("Enter a string: ")
list_of_letters = list(user_input)
reverse_string(user_input)

user_input = (input("Enter a integer: "))
reverse_integer(user_input)

user_input = input("Enter a string: ")
user_input = user_input.replace(" ", "")
list_of_letters = list(user_input)
uppercase_count(list_of_letters)
lowercase_count(list_of_letters)
first_vowel_position(list_of_letters)
character_ascii(list_of_letters)

avg_one_to_hundred()

user_input = input("Enter a number to check palindrome: ")
palindrome_number(user_input)

user_input = input("Enter a string to check palindrome: ")
palindrome_string(user_input)

user_input = int(input("Enter a integer: "))
divisors(user_input)

number = int(input("Enter a number: "))
exp = int(input("Enter the power raside to: "))
power(number, exp)

user_input = input("Enter a sentence: ")
list_of_letters = list(user_input)
number_of_words(list_of_letters)

user_input = (input("Enter a integer: "))
sum_of_even_digits(user_input)
sum_of_odd_digits(user_input)

prime_numbers()

fibonacci_sequence()

user_input = input("Enter a binary number: ")
binary_to_decimal(user_input)

user_input = int(input("Enter a decimal number: "))
decimal_to_binary(user_input)

File: Java   Assignments/WeekendExercise/Task.java
import java.util.*;

public class Task {
    
    public static void reverseString(String userInput) {
        char[] letters = userInput.toCharArray();
        for (int index = letters.length - 1; index >= 0; index--) {
            System.out.print(letters[index]);
        }
        System.out.println();
    }

    public static void reverseInteger(String userInput) {
        char[] digits = userInput.toCharArray();
        for (int index = digits.length - 1; index >= 0; index--) {
            System.out.print(digits[index]);
        }
        System.out.println();
    }

    public static void uppercaseCount(String userInput) {
        int count = 0;
        for (char letters : userInput.toCharArray()) {
            if (Character.isUpperCase(letters)) count++;
        }
        System.out.println("Number of uppercase letters: " + count);
    }

    public static void lowercaseCount(String userInput) {
        int count = 0;
        for (char letters : userInput.toCharArray()) {
            if (Character.isLowerCase(letters)) count++;
        }
        System.out.println("Number of lowercase letters: " + count);
    }

    public static void firstVowelPosition(String userInput) {
        String vowels = "aeiou";
        char[] listOfLetters = userInput.toLowerCase().toCharArray();
        for (int index = 0; index < listOfLetters.length; index++) {
            if (vowels.contains(listOfLetters[index] + "")) {
                System.out.println("The position of the first vowel in the string is " + (index + 1));
                break;
            }
        }
    }

    public static void avgOneToHundred() {
        int sum = 0;
        for (int number = 1; number <= 100; number++){
            sum += number;
        } 
        System.out.println("Average of numbers from 1 to 100 = " + (sum / 100.0));
    }

    public static void divisors(int userInput) {
        int count = 0;
        System.out.print("Divisors: ");
        for (int index = 1; index <= userInput; index++) {
            if (userInput % index == 0) {
                System.out.print(index + ", ");
                count++;
            }
        }
        System.out.println("\nNumber of divisors = " + count);
    }

    public static void power(int number, int exp) {
        int result = 1;
        for (int index = 0; index < exp; index++) {
            result *= number;
        }
        System.out.println(result);
    }
    
    public static void palindromeNumber(String userInput) {
        char[] listOfDigits = userInput.toCharArray();
        char[] reversedList = new char[listOfDigits.length];
        
        for (int index = 0; index < listOfDigits.length; index++) {
            reversedList[index] = listOfDigits[listOfDigits.length - 1 - index];
        }
        
        if (Arrays.equals(listOfDigits, reversedList)) {
                System.out.println("The number is a palindrome");
        } else {
            System.out.println("The number is not a palindrome");
        }
    }
    
    public static void palindromeString(String userInput) {
        char[] listOfLetters = userInput.toCharArray();
        char[] reversedList = new char[listOfLetters.length];
        
        for (int index = 0; index < listOfLetters.length; index++) {
            reversedList[index] = listOfLetters[listOfLetters.length - 1 - index];
        }
        
        if (Arrays.equals(listOfLetters, reversedList)) {
                System.out.println("The string is a palindrome");
        } else {
            System.out.println("The string is not a palindrome");
        }
    }

    public static void numberOfWords(String userInput) {
        int numberOfWords = 1;
        for (int index = 0; index < userInput.length(); index++) {
            if (userInput.charAt(index) == ' ') {
                numberOfWords++;
            }
        }
        System.out.println(numberOfWords);
    }

    public static void sumOfEvenDigits(String userInput) {
        int sumOfEven = 0;
        int digit;
        System.out.println("HERE");
        for (int index = 0; index < userInput.length(); index++) {
            digit = (userInput.charAt(index) - '0') ;
            if (digit % 2 == 0) {
                sumOfEven += digit;
            }
        }
        System.out.println("Sum of all even digits is = " + sumOfEven);
    }//Fix, Starts at 48
    
    public static void sumOfOddDigits(String userInput) {
        int sumOfOdd = 0;
        for (int index = 0; index < userInput.length(); index++) {
            int digit = userInput.charAt(index) - '0';
            if (digit % 2 != 0) {
                sumOfOdd += digit ;
            }
        }
        System.out.println("Sum of all odd digits is = " + sumOfOdd);
    }//Fix, Starts at 48

    static void primeNumbers() {
        int count = 0;
        for (int number = 2; number <= 100; number++) {
            boolean isPrime = true;
            for (int index = 2; index < number; index++) {
                if (number % index == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                System.out.print(number + " ");
                count++;
            }
        }
        System.out.println("\nTotal primes = " + count);
    }

    static void fibonacciSequence() {
        int firtsNumber = 1, secondNumber = 1;
        for (int index = 0; index < 20; index++) {
            System.out.print(firtsNumber + " ");
            int temp = secondNumber;
            secondNumber = firtsNumber + secondNumber;
            firtsNumber = temp;
        }
        System.out.println();
    }

    static void binaryToDecimal(String userInput) {
        int decimal = 0;
        int power = 0;
        for (int index = userInput.length() - 1; index >= 0; index--) {
            decimal += (userInput.charAt(index) - '0') * Math.pow(2, power++);
        }
        System.out.println("Decimal value = " + decimal);
    }

    static void decimalToBinary(int userInput) {
        String binary = "";
        while (userInput > 0) {
            binary = (userInput % 2) + binary;
            userInput /= 2;
        }
        System.out.println("Binary value = " + binary);
    }

    public static void main(String[] args) {
        Scanner userInputScanner = new Scanner(System.in);
        
        System.out.print("Enter a string: ");
        String userInput = userInputScanner.nextLine();
        reverseString(userInput);
        
        System.out.print("Enter a integer: ");
        userInput = userInputScanner.nextLine();
        reverseInteger(userInput);
        
        System.out.print("Enter a string: ");
        userInput = userInputScanner.nextLine();
        userInput = userInput.replace(" ", "");
        uppercaseCount(userInput);
        lowercaseCount(userInput);
        firstVowelPosition(userInput);
      
        
        avgOneToHundred();
        
        System.out.print("Enter a number to check palindrome: ");
        userInput = userInputScanner.nextLine();
        palindromeNumber(userInput);
        
        System.out.print("Enter a string to check palindrome: ");
        userInput = userInputScanner.nextLine();
        palindromeString(userInput);
        
        System.out.print("Enter a integer: ");
        int numInput = userInputScanner.nextInt();
        divisors(numInput);
        
        System.out.print("Enter a number: ");
        int number = userInputScanner.nextInt();
        System.out.print("Enter the power raised to: ");
        int exp = userInputScanner.nextInt();
        power(number, exp);
        userInputScanner.nextLine();
        
        System.out.print("Enter a sentence: ");
        userInput = userInputScanner.nextLine();
        numberOfWords(userInput);
        
        System.out.print("Enter a integer: ");
        userInput = userInputScanner.nextLine();
        sumOfEvenDigits(userInput);
        sumOfOddDigits(userInput);
        
        primeNumbers();
        
        fibonacciSequence();
        
        System.out.print("Enter a binary number: ");
        userInput = userInputScanner.nextLine();
        binaryToDecimal(userInput);
        
        System.out.print("Enter a decimal number: ");
        numInput = userInputScanner.nextInt();
        decimalToBinary(numInput);

    }
}